modules/blocks.php

```diff
@@ -3,299 +3,22 @@
  * Load code specific to Gutenberg blocks which are not tied to a module.
  * This file is unusual, and is not an actual `module` as such.
  * It is included in ./module-extras.php
- */
-
-/**
- * Map block.
  *
- * @since 6.8.0
+ * @package Jetpack
  */
-jetpack_register_block(
-	'map',
-	array(
-		'render_callback' => 'jetpack_map_block_load_assets',
-	)
-);
 
 /**
- * Map block registration/dependency declaration.
- *
- * @param array $attr - Array containing the map block attributes.
- * @param string $content - String containing the map block content.
- *
- * @return string
+ * Load all blocks inside the modules/blocks folder.
  */
-function jetpack_map_block_load_assets( $attr, $content ) {
-	$dependencies = array(
-		'lodash',
-		'wp-element',
-		'wp-i18n',
-	);
-
-	$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
-
-	Jetpack_Gutenberg::load_assets_as_required( 'map', $dependencies );
-
-	return preg_replace( '/<div /', '<div data-api-key="' . esc_attr( $api_key ) . '" ', $content, 1 );
-}
-
+function jetpack_load_blocks() {
+	$blocks_include = array();
 
-/**
- * Tiled Gallery block. Depends on the Photon module.
- *
- * @since 6.9.0
- */
-if (
-	( defined( 'IS_WPCOM' ) && IS_WPCOM ) ||
-	class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
-) {
-	jetpack_register_block(
-		'tiled-gallery',
-		array(
-			'render_callback' => 'jetpack_tiled_gallery_load_block_assets',
-		)
-	);
-
-	/**
-	 * Tiled gallery block registration/dependency declaration.
-	 *
-	 * @param array $attr - Array containing the block attributes.
-	 * @param string $content - String containing the block content.
-	 *
-	 * @return string
-	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
-		$dependencies = array(
-			'lodash',
-			'wp-i18n',
-			'wp-token-list',
-		);
-		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery', $dependencies );
-
-		/**
-		 * Filter the output of the Tiled Galleries content.
-		 *
-		 * @module tiled-gallery
-		 *
-		 * @since 6.9.0
-		 *
-		 * @param string $content Tiled Gallery block content.
-		 */
-		return apply_filters( 'jetpack_tiled_galleries_block_content', $content );
+	foreach ( Jetpack::glob_php( dirname( __FILE__ ) . '/blocks' ) as $file ) {
```

Comment:
Noting that I'll need to update this as this is not available on WordPress.com today.

Reply:
Here is the approach I took instead: 6dc1f24abc8292525699b9fae9c0d541ecbd64fe